staging/src/k8s.io/apiserver/pkg/server/types/hooks.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
```

Comment:
why not use the existing k8s.io/apiserver/pkg/server/hooks.go

Reply:
I moved the type definitions to a sub package to avoid import cycle.

Reply:
Oh, I see.